ui/src/containers/AppVersionsTableContainer.js

```diff
@@ -9,20 +9,42 @@ import './TableContainer.css';
 import config from '../config/config';
 
 export class AppVersionsTableContainer extends React.Component {
+  state = {
+    canUpdateComponent: true
+  }
+
+  shouldComponentUpdate () {
+    return this.state.canUpdateComponent;
```

Comment:
are we using this correctly? docs & examples I'm looking at tend to look like:
```
shouldComponentUpdate(nextProps, nextState) {
  return nextProps.isFavourite != this.props.isFavourite;
}
```
Links I was reading: https://reactjs.org/docs/react-component.html#shouldcomponentupdate, https://medium.freecodecamp.org/react-shouldcomponentupdate-demystified-c5d323099ef6

Reply:
I first tried this way and I don't think it would work for what we are trying to do. First off, we are dealing with a dynamic list of versions, where two properties from each can be updated. We could either:

a) do a binary comparison of the `this.props.appVersions` and `nextProps.appVersions`, but there are other ways that this prop can be altered (sorting).
b) on every `shouldComponentUpdate`, loop through the array and compare the values to the current state. They would have to be mapped to each other by their ID. I'd imagine doing a `.forEach` and a `.find` on every update to compare values would be more costly than the current implementation.
c) What is there currently just seems simpler than b).

Reply:
Yes, agree - had forgotten we were using `appVersions` to store the sort order. This implementation is great :+1: